contribute/design/1.x/csi/20190107-cStor-volume-resize.md

```diff
@@ -0,0 +1,357 @@
+1. Edit the pvc spec capacity using `kubectl edit pvc <pvc_name>`.
+
+Consider following PVC as example for CSI cStor Volume resize
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"v1","kind":"PersistentVolumeClaim","metadata":{"annotations":{},"name":"csi-vol-claim","namespace":"default"},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"5Gi"}},"storageClassName":"openebs-csi-sc"}}
+    pv.kubernetes.io/bind-completed: "yes"
+    pv.kubernetes.io/bound-by-controller: "yes"
+    volume.beta.kubernetes.io/storage-provisioner: cstor.csi.openebs.io
+  creationTimestamp: "2019-12-23T13:11:33Z"
+  finalizers:
+  - kubernetes.io/pvc-protection
+  name: csi-vol-claim
+  namespace: default
+  resourceVersion: "1387"
+  selfLink: /api/v1/namespaces/default/persistentvolumeclaims/csi-vol-claim
+  uid: a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 5Gi
+  storageClassName: openebs-csi-sc
+  volumeMode: Filesystem
+  volumeName: pvc-a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+status:
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 5Gi
+  phase: Bound
+```
+
+After updating spec.capacity in above PVC --- then PVC will look following:
+```yaml
+piVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"v1","kind":"PersistentVolumeClaim","metadata":{"annotations":{},"name":"csi-vol-claim","namespace":"default"},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"5Gi"}},"storageClassName":"openebs-csi-sc"}}
+    pv.kubernetes.io/bind-completed: "yes"
+    pv.kubernetes.io/bound-by-controller: "yes"
+    volume.beta.kubernetes.io/storage-provisioner: cstor.csi.openebs.io
+  creationTimestamp: "2019-12-23T13:11:33Z"
+  finalizers:
+  - kubernetes.io/pvc-protection
+  name: csi-vol-claim
+  namespace: default
+  resourceVersion: "7804"
+  selfLink: /api/v1/namespaces/default/persistentvolumeclaims/csi-vol-claim
+  uid: a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+  storageClassName: openebs-csi-sc
+  volumeMode: Filesystem
+  volumeName: pvc-a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+status:
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 5Gi
+  conditions:
+  - lastProbeTime: null
+    lastTransitionTime: "2019-12-23T13:59:08Z"
+    message: Waiting for filesystem resize.
+    status: "True"
+    type: FileSystemResizePending
+  phase: Bound
+```
+In the above PVC spec capacity is updated to 10Gi but status capacity is 5Gi(Which
+means resize is in progress).
+
+### Implementation Details
+
+The sections below can be assumed to be specific to `cStor` unless mentioned
+otherwise.
+
+### Custom Resources used to resize cStor volume
```

Comment:
This block appears out of no-where. Move this section below the implementation workflow. And create a link to this section, in the note that is mentioning to refer to this.

Reply:
Done